### PR TITLE
Refactor metadata

### PR DESCRIPTION
**Pull request by jrussell25**
This PR updates `load_mm_frames` to also read the metadata and assign coords to the dimensions where it is appropriate.

More importnantly, it changes the default naming scheme for DataArrays to the `STCZYX` convention. Subsequently it also allows more argument passing into `squash_stack` for increased flexibility.

### CODE REVIEW

File: microutil/preprocess.py [hunk 1]
```diff
@@ -110,40 +114,45 @@ def xarray_plls(log_power_spec, logR):
     )
 
 
-def squash_zstack(data, squash_fn="max", bf_name="BF", channel_name="channel", transpose=True):
+def squash_zstack(
+    data,
+    squash_fn="max",
```
Comment (ianhi) [inline microutil/preprocess.py:119]:
```suggestion
    squash_fn="mean",
```

nows as good a time as ever to swap the default?

Comment (ianhi) [reply microutil/preprocess.py:119]: Also probs a good time to add a real docstring

File: microutil/loading.py [hunk 2]
```diff
@@ -160,6 +201,16 @@ def load_mm_frames(data_dir, glob_pattern=None, chunkby_dims=['C', 'Z']):
     chunks = np.expand_dims(chunks, tuple(range(-1, -len(chunkby_dims) - 3, -1)))
 
     d_data = da.block(chunks.tolist())
+
     x_data = xr.DataArray(da.block(chunks.tolist()), dims=group_dims + chunkby_dims + ['Y', 'X'])
 
+    with open(position_dirs[0] + "/metadata.txt") as f:
+        meta = json.load(f)
```
Comment (ianhi) [inline microutil/loading.py:208]: What do you think about adding an escape hatch of just providing a path to the metadata file. LIke say pos0 is borked so you only want to load pos1-3, but as I understand it the metadata file only exists in the pos0 folder?

**Comment by ianhi** [comment]
looking great. We should however definitely update our test case to also include the metadata

File: microutil/loading.py (hunk 2 above)
Comment (jrussell25) [reply microutil/loading.py:208]: the metadata should exist in each position directory. so in practice this just uses the first one not necessarily pos0.

Maybe the move is to have a kwarg in the loading files that allow you to pass a dict directly. So if the metadata is borked you just make the dict yourself and zoop it through.

File: microutil/preprocess.py (hunk 1 above)
Comment (jrussell25) [reply microutil/preprocess.py:119]: Great call swapping the default i meant to and forgot.

**Pull request merged**